module-apps/application-settings/widgets/network/ApnInputWidget.hpp

```diff
@@ -16,12 +16,14 @@ namespace gui
         settingsInternals::ListItemName listItemName;
 
       public:
-        ApnInputWidget(settingsInternals::ListItemName listItemName,
-                       std::function<void(const UTF8 &text)> navBarTemporaryMode   = nullptr,
-                       std::function<void()> navBarRestoreFromTemporaryMode        = nullptr,
-                       std::function<void()> selectSpecialCharacter                = nullptr,
-                       std::function<void(const std::string &text)> contentChanged = nullptr,
-                       unsigned int lines                                          = 1);
+        explicit ApnInputWidget(
+            settingsInternals::ListItemName listItemName,
+            const std::function<void(const UTF8 &)> &navBarTemporaryMode                       = nullptr,
+            const std::function<void()> &navBarRestoreFromTemporaryMode                        = nullptr,
+            const std::function<void()> &selectSpecialCharacter                                = nullptr,
+            const std::function<void(std::function<void()> restoreFunction)> &restoreInputMode = nullptr,
```

Comment:
`std::function` inside an `std::function`? Looks a bit suspicious :)

Reply:
It's just a function that takes function as an argument; I needed to do it this way, so that the function from inside InputMode (the one passed as an argument `restoreFunction`) can be called from the outside, by the timer. The problem is that in our design every timer has to be connected with some application or service instance, so either I'd need to pass pointer to parent application to every window that implements InputMode (and from there to the InputMode itself) or do it as I did, with additional callback that provides the function to be called by timer as an argument. Or maybe there's another solution that I've missed...?

Reply:
OK, seems reasonable - that can really be a separate architectural fault